app-modules/interaction/src/Imports/InteractionsImporter.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+namespace Assist\Interaction\Imports;
+
+use App\Models\Import;
+use App\Imports\Importer;
+use Illuminate\Support\Str;
+use Assist\Prospect\Models\Prospect;
+use Illuminate\Database\Eloquent\Builder;
+use Assist\AssistDataModel\Models\Student;
+use Assist\Interaction\Models\Interaction;
+use Assist\Interaction\Models\InteractionType;
+use Assist\Interaction\Models\InteractionDriver;
+use Assist\Interaction\Models\InteractionStatus;
+use Assist\Interaction\Models\InteractionOutcome;
+use Assist\Interaction\Models\InteractionCampaign;
+use Assist\Interaction\Models\InteractionRelation;
+use App\Filament\Actions\ImportAction\ImportColumn;
+use Assist\Interaction\Models\InteractionInstitution;
+
+class InteractionsImporter extends Importer
+{
+    /**
+     * @inheritDoc
+     */
+    public static function getColumns(): array
+    {
+        return [
+            ImportColumn::make('interactable')
+                ->relationship(
+                    resolveUsing: function (mixed $state) {
+                        $type = str($state)->before(':');
+                        $value = str($state)->after(':');
+
+                        return match ($type->toString()) {
+                            'prospect' => Prospect::query()
+                                ->when(
+                                    str($value)->isUuid(),
+                                    fn (Builder $query) => $query->whereKey($value),
+                                    fn (Builder $query) => $query->where('email', $value),
+                                )
+                                ->first(),
+                            'student' => Student::query()
+                                ->when(
+                                    str($value)->isUuid(),
+                                    fn (Builder $query) => $query->whereKey($value),
+                                    fn (Builder $query) => $query->where('email', $value),
+                                )
+                                ->first(),
+                        };
+                    },
+                )
+                ->requiredMapping()
+                ->rules(
+                    [
+                        'starts_with:prospect:,student:',
+                    ]
+                )
+                ->example(fn (): string => 'student:' . Student::query()->value('email') ?? fake()->safeEmail()),
```

Comment:
I believe this will fail in production because `composer install --no-dev` will not install faker.